scheduler/graphql_mid/schema.py

```diff
@@ -83,6 +87,32 @@ def change_origin(self, new_origin: SOrigin, mode: SchedulerModes) -> ChangeOrig
         sources.set_origin(new_origin)
         return ChangeOriginSuccess(from_origin=old, to_origin=str(new_origin))
 
+    @strawberry.mutation
+    def add_weather_change(self, new_weather_change: NewWeatherChangeInput) -> EventsAddedResponse:
+        try:
+            new_cc = new_weather_change.new_CC
+            new_sb = new_weather_change.new_SB
+            new_wv = new_weather_change.new_WV
+            new_iq = new_weather_change.new_IQ
+            c = Conditions(cc=CloudCover[new_cc.name] if new_cc else None,
+                           sb=SkyBackground[new_sb.name] if new_sb else None,
+                           wv=WaterVapor[new_wv.name] if new_wv else None,
+                           iq=ImageQuality[new_iq.name] if new_iq else None)
+            wc = WeatherChange(event_id=uuid.uuid4(),
+                               start=new_weather_change.start,
+                               reason=new_weather_change.reason,
+                               site=new_weather_change.site,
+                               new_conditions=c)
+            event_queue.add_event(time_handler.dt2night_index(new_weather_change.start), new_weather_change.site, wc)
+        except KeyError:
+            return EventsAddedError(error='Weather Condition not valid!')
+        else:
+            return EventsAddedSuccess(added_event=f'Weather change added at {new_weather_change.start.timestamp()}')
+
+    @strawberry.mutation
+    def remove_weather_change(self, night: int, site: Site, event_id: uuid.UUID) -> EventRemovedResponse:
+        return EventRemovedResponse(success=event_queue.remove_event(NightIndex(night), site, event_id))
```

Comment:
This seems like an interesting addition and I'm curious about the use cases: when would we remove a weather change? (It's not a bad idea at all. I'm just wondering what you're thinking.)

Reply:
To remove already created events and change the validation.